src/Sky.cc

```diff
@@ -154,6 +157,18 @@ void Sky::SetCloudAmbient(const ignition::math::Color &_ambient)
   this->dataPtr->cloudAmbient = _ambient;
 }
 
+//////////////////////////////////////////////////
+std::string Sky::CubemapUri() const
```

Comment:
Can we prevent a copy here?

```suggestion
const std::string &Sky::CubemapUri() const
```

Reply:
Oh this needs to change on the header too, CI is failing to compile.

Reply:
[fc0cf3f](https://github.com/gazebosim/sdformat/pull/1037/commits/fc0cf3fb2858be09964412971e9ac38a676c4874)

Reply:
I think that might have been some sort of infrastructure issue. It's complaining about `readXMLStream`, which is not touched by this PR. Also if you look at the [console logs](https://build.osrfoundation.org/job/sdformat-abichecker-any_to_any-ubuntu_auto-amd64/2252/console), I don't think it really ran. I think it reported the result of another job.

Since this is a new function, I wouldn't think changing the return type has affects ABI compatibility. I would recommend going back to returning a const ref.

Reply:
Hmm okay. I'll revert back and see what it says. I reran the ABI check 3 times and it gave me the same thing, and the previous jobs were passing, so it seemed like it was just failing here. Maybe now it is better :sweat_smile:

Reply:
Ooh it's passing now!

Reply:
Good to merge?